src/Serilog.Sinks.MSSqlServer/Configuration/Implementations/Microsoft.Extensions.Configuration/MicrosoftExtensionsSinkOptionsProvider.cs

```diff
@@ -33,7 +34,11 @@ private static void ReadBatchSettings(IConfigurationSection config, MSSqlServerS
             SetProperty.IfNotNull<string>(config["batchPeriod"], val => sinkOptions.BatchPeriod = TimeSpan.Parse(val, CultureInfo.InvariantCulture));
             SetProperty.IfNotNull<bool>(config["eagerlyEmitFirstEvent"], val => sinkOptions.EagerlyEmitFirstEvent = val);
         }
-
+        private static void ReadRetentionSettings(IConfigurationSection config, MSSqlServerSinkOptions sinkOptions)
+        {
+            SetProperty.IfNotNull<string>(config["retentionPeriod"], val => sinkOptions.RetentionPeriod  = TimeSpan.Parse(val, CultureInfo.InvariantCulture));
+            SetProperty.IfNotNull<string>(config["cleaningInterval"], val => sinkOptions.CleaningInterval = TimeSpan.Parse(val, CultureInfo.InvariantCulture));
```

Comment:
Add range valiadation so that cleaningInverval cannot be set too short so the delete does not run on every batch.